src/message/commands/privmsg.rs

```diff
@@ -66,6 +70,15 @@ impl TryFrom<IRCMessage> for PrivmsgMessage {
 
         let (message_text, is_action) = source.try_get_message_text()?;
 
+        let (is_redeem, custom_reward_id) = {
+            let msg_id = source.try_get_optional_tag_value("msg-id")?.to_owned();
+            let custom_reward_id = source.try_get_optional_tag_value("custom-reward-id")?.to_owned();
+
+            let is_redeem = custom_reward_id.is_some() || msg_id == Some(String::from("skip-subs-mode-message")) || msg_id == Some(String::from("highlighted-message"));
```

Comment:
Is there some kind of documentation for these tags and tag values? I couldn't find any straight away

Reply:
https://github.com/tmijs/tmi.js/blob/cee8a42b9923ec4e947036018e6da6bf08ed847f/lib/client.js#L1083

I found only here the correct reference

Reply:
And here https://github.com/tmijs/tmi.js/blob/fae890153e8b3a2ab6bd56763f684606e0ebdb3c/test/events.js#L636